File: php/testcode/unsafereflect_013.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_reflect_match_dispatch
class ExportAction013 { public function run(): string { return 'exported'; } }
class ImportAction013 { public function run(): string { return 'imported'; } }
class ReportAction013 { public function run(): string { return 'reported'; } }

function unsafereflect013(BenchmarkRequest $req): BenchmarkResponse {
    $action = $req->param('action');
    $handler = match($action) { // vuln-code-snippet safe-line php_reflect_match_dispatch
        'export' => new ExportAction013(),
        'import' => new ImportAction013(),
        'report' => new ReportAction013(),
        default => null,
    };
    if ($handler === null) {
        return BenchmarkResponse::badRequest('Unknown action');
    }
    return BenchmarkResponse::ok($handler->run());
}
// vuln-code-snippet end php_reflect_match_dispatch

File: ruby/testcode/unsafereflect_013.rb
require_relative 'shared'

# vuln-code-snippet start ruby_reflect_autoload_trigger
def load_and_create(req)
  gem_name = req.param('gem')
  class_name = req.param('class')
  require gem_name
  klass = Object.const_get(class_name) # vuln-code-snippet vuln-line ruby_reflect_autoload_trigger
  BenchmarkResponse.ok(klass.new.to_s)
end
# vuln-code-snippet end ruby_reflect_autoload_trigger
